crates/polars-core/src/datatypes/dtype.rs

```diff
@@ -572,6 +572,52 @@ impl DataType {
         }
     }
 
+    /// Try to get the maximum value for this datatype.
+    pub fn max(&self) -> PolarsResult<Scalar> {
+        use DataType::*;
+        let v = match self {
+            #[cfg(feature = "dtype-i8")]
+            Int8 => Scalar::from(i8::MAX),
+            #[cfg(feature = "dtype-i16")]
+            Int16 => Scalar::from(i16::MAX),
+            Int32 => Scalar::from(i32::MAX),
+            Int64 => Scalar::from(i64::MAX),
+            #[cfg(feature = "dtype-u8")]
+            UInt8 => Scalar::from(u8::MAX),
+            #[cfg(feature = "dtype-u16")]
+            UInt16 => Scalar::from(u16::MAX),
+            UInt32 => Scalar::from(u32::MAX),
+            UInt64 => Scalar::from(u64::MAX),
+            Float32 => Scalar::from(f32::INFINITY),
+            Float64 => Scalar::from(f64::INFINITY),
+            dt => polars_bail!(ComputeError: "cannot determine upper bound for dtype `{}`", dt),
```

Comment:
I think here we can use something like [`NaiveDate::MAX()`](https://docs.rs/chrono/latest/chrono/naive/struct.NaiveDate.html#associatedconstant.MAX) and `MIN` from chrono here to get date limits, and likewise for `NaiveDateTime`. As an example, the naive date max value is `95026236` and we can see we hit that limit here as well:

```python
import polars as pl

pl.Series([95026236]).cast(pl.Date)  # works
pl.Series([95026237]).cast(pl.Date)  # PanicException: out-of-range date
```

Reply:
Good point. Perhaps could be discussed in a follow-up issue?

I don't know if we should include chrono limitations here or not.